dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/update/postprocess/PublishEventPostProcessor.java

```diff
@@ -69,15 +64,8 @@ public void process( final Event event, final WorkContext ctx )
 
         if ( !ctx.getImportOptions().isSkipNotifications() && isLinkedWithRuleVariable )
         {
-            ProgramStageInstance programStageInstance = getProgramStageInstance( ctx, event );
             ctx.getServiceDelegator().getApplicationEventPublisher().publishEvent(
-                new DataValueUpdatedEvent( this, programStageInstance.getUid() ) );
+                new DataValueUpdatedEvent( this, event.getId() ) );
         }
     }
-
```

Comment:
It seems unnecessary to map to `ProgramStageInstance` when we just need the ID, which we can just get from `Event`. Correct @maikelarabori @zubaira ?